gtsam/hybrid/DCGaussianMixtureFactor.h

```diff
@@ -69,6 +67,28 @@ class DCGaussianMixtureFactor : public Factor {
     return 0;
   }
 
+  // TODO(dellaert): Does not make sense
+  double error(const gtsam::Values& continuousVals,
+               const gtsam::DiscreteValues& discreteVals) const override {
+    return 0;
+  }
+
+  // TODO(dellaert): Does not make sense
+  GaussianFactor::shared_ptr linearize(
+      const gtsam::Values& continuousVals,
+      const DiscreteValues& discreteVals) const override {
+    return factors_[0];
+  }
+
+  /// Return this linear factor as a DCFactor::shared_ptr
```

Comment:
Also does not make sense: factor is already linear.

Reply:
Yeah the issues is that without an implementation of that method, `DCGaussianMixtureFactor` is an abstract type so we cannot initialize it in DCMixtureFactor. :(

Reply:
Maybe we should have a DCFactor and a corresponding DCGaussianFactor?
Or we can just remove the error and linearize methods from the DCFactor class.

Reply:
Just check its type using dynamic pointer cast. Or have a method `isLinear`.

Reply:
Would we be checking just for `DCGaussianMixtureFactor`? Asking since we want to keep the possibility of adding other `DCGaussianXFactor`s in the future I guess?

Reply:
Also, since DCGaussianMixtureFactor inherits from DCFactor, we kinda need this method no?